packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -0,0 +1,499 @@
+    };
+
+    /**
+     * Handled props instantiation
+     */
+    protected handledProps: HandledProps<SelectHandledProps> = {
+        isMenuOpen: void 0,
+        disabled: void 0,
+        displayStringFormatter: void 0,
+        form: void 0,
+        labelledBy: void 0,
+        multiselectable: void 0,
+        trigger: void 0,
+        required: void 0,
+        managedClasses: void 0,
+        selectedItems: void 0,
+        defaultSelection: void 0,
+        onValueChange: void 0,
+        placeholder: void 0,
+        autoFocus: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement> = React.createRef<
+        HTMLDivElement
+    >();
+
+    /**
+     * constructor
+     */
+    constructor(props: SelectProps) {
+        super(props);
+
+        let initialSelection: ListboxItemProps[];
+        if (this.props.selectedItems !== undefined) {
+            initialSelection = Listbox.getListboxItemDataFromIds(
+                this.props.selectedItems,
+                this.props.children
+            );
+        } else {
+            initialSelection = Listbox.getListboxItemDataFromIds(
+                this.props.defaultSelection,
+                this.props.children
+            );
+        }
+
+        if (!this.props.multiselectable && initialSelection.length > 1) {
+            initialSelection = initialSelection.slice(0, 1);
+        }
+
+        this.state = {
+            selectedItems: initialSelection,
+            value: "",
+            displayString: "",
+            isMenuOpen: this.validateMenuState(false),
+        };
+    }
+
+    public componentDidUpdate(prevProps: SelectProps): void {
+        if (prevProps.selectedItems !== this.props.selectedItems) {
+            const newSelection: ListboxItemProps[] = Listbox.getListboxItemDataFromIds(
+                this.props.selectedItems,
+                this.props.children
+            );
+            this.updateSelection(newSelection);
+        }
+        if (prevProps.multiselectable !== this.props.multiselectable) {
+            if (
+                this.props.multiselectable === false &&
+                this.state.selectedItems.length > 1
+            ) {
+                this.updateSelection([this.state.selectedItems[0]]);
+            }
+            this.toggleMenu(this.checkPropsForMenuState());
+        }
+        if (prevProps.isMenuOpen !== this.props.isMenuOpen) {
+            this.toggleMenu(this.checkPropsForMenuState());
+        }
+    }
+
+    public componentDidMount(): void {
+        let initialSelection: ListboxItemProps[];
+        if (this.props.selectedItems !== undefined) {
+            initialSelection = Listbox.getListboxItemDataFromIds(
+                this.props.selectedItems,
+                this.props.children
+            );
+        } else {
+            initialSelection = Listbox.getListboxItemDataFromIds(
+                this.props.defaultSelection,
+                this.props.children
+            );
+        }
```

Comment:
nit: I would prefer this reads the same way I suggested above:

```
initialSelection = Listbox.getListboxItemDataFromIds(
    this.props.selectedItems !== undefined ? this.props.selectedItems : this.props.defaultSelection,
    this.props.children
);
```

Reply:
sure